src/sqlfluff/core/rules/base.py

```diff
@@ -221,6 +228,34 @@ class RuleContext:
     path: Optional[pathlib.Path]
     templated_file: Optional[TemplatedFile]
 
+    @cached_property
+    def surrogates(self):
+        """Returns a Surrogates object that simplifies writing rules."""
+        return Surrogates(self)
+
+
+class Surrogates:
+    """Provides access to surrogate objects that simplify writing rules."""
+
+    def __init__(self, context: RuleContext):
+        self.context = context
+
+    @cached_property
+    def segment(self) -> "Segments":
+        """Returns a Segments object for context.segment."""
+        # Import here to avoid circular dependency.
+        from sqlfluff.core.rules.surrogates import Segments
```

Comment:
Feels a little weird to have to do this. As far as I can see the only thing we're importing from this file in `sqlfluff.core.rules.surrogates` is `LintFix`. Feels like `LintFix` and `LintResult` should be in their own file anyway. Would moving those to a separate file stop this?

Reply:
Probably. The circular dependency is between RuleContext -> Segments -> LintFix.

Reply:
Making this change now. Seems to fix the circular dependency -- great suggestion!